bpf/bpf_lxc.c

```diff
@@ -1029,27 +1025,15 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 		if (ct_status == CT_REPLY || ct_status == CT_RELATED)
 			goto skip_egress_gateway;
 
-		egress_gw_policy = lookup_ip4_egress_gw_policy(ip4->saddr, ip4->daddr);
-		if (!egress_gw_policy)
-			goto skip_egress_gateway;
-
-		/* If the gateway node is the local node, then just let the
-		 * packet go through, as it will be SNATed later on by
-		 * handle_nat_fwd().
-		 */
-		gateway_node_ep = __lookup_ip4_endpoint(egress_gw_policy->gateway_ip);
-		if (gateway_node_ep && (gateway_node_ep->flags & ENDPOINT_F_HOST))
-			goto skip_egress_gateway;
+		if (egress_gw_request_needs_redirect(ip4, &tunnel_endpoint)) {
```

Comment:
```suggestion
		if (!egress_gw_request_needs_redirect(ip4, &tunnel_endpoint))
			goto skip_egress_gateway;
```

Reply:
ack, nice! Staged them in my `for-egressgw` branch for next time :+1:.